tests/api/runBlock.js

```diff
@@ -0,0 +1,123 @@
+const { promisify } = require('util')
+const tape = require('tape')
+const Block = require('ethereumjs-block')
+const Transaction = require('ethereumjs-tx')
+const Common = require('ethereumjs-common')
+const util = require('ethereumjs-util')
+const runBlock = require('../../lib/runBlock')
+const StateManager = require('../../lib/stateManager')
+const runTx = require('../../lib/runTx')
+const testData = require('./testdata.json')
+const { createGenesis, createAccount, setupVM } = require('./utils')
+const { setupPreConditions } = require('../util')
+
+function setup (vm = null) {
+  // Create a mock, if no real VM object provided.
+  if (vm === null) {
+    const stateManager = new StateManager()
+    vm = {
+      stateManager,
+      emit: (e, val, cb) => cb(),
+      populateCache: stateManager.warmCache.bind(stateManager),
+      runTx: (opts, cb) => cb(new Error('test')),
+      runCall: (opts, cb) => cb(new Error('test')),
+      _common: new Common('mainnet', 'byzantium')
+    }
+  }
+
+  return {
+    vm,
+    data: testData,
+    p: {
+      runBlock: promisify(runBlock.bind(vm)),
+      putAccount: promisify(vm.stateManager.putAccount.bind(vm.stateManager)),
+      cacheFlush: promisify(vm.stateManager.cache.flush.bind(vm.stateManager.cache))
+    }
+  }
+}
+
+tape('runBlock', async (t) => {
+  const suite = setup()
+
+  t.test('should fail without params', async (st) => {
+    suite.p.runBlock()
+      .then(() => st.fail('should have returned error'))
+      .catch((e) => st.ok(e.message.includes('invalid input'), 'correct error'))
+
+    st.end()
+  })
+
+  t.test('should fail without opts', async (st) => {
+    suite.p.runBlock({})
+      .then(() => st.fail('should have returned error'))
+      .catch((e) => st.ok(e.message.includes('invalid input'), 'correct error'))
+
+    st.end()
+  })
+
+  t.test('should fail when runTx fails', async (st) => {
+    const genesis = createGenesis()
+    const block = new Block(util.rlp.decode(suite.data.blocks[0].rlp))
+    await suite.p.runBlock({ block, root: genesis.header.stateRoot })
+      .then(() => t.fail('should have returned error'))
+      .catch((e) => t.equal(e.message, 'test'))
+
+    st.end()
```

Comment:
Not seeing this instantaneously, why is ``runTx`` failing here? Maybe this is more obvious then I get right now, but eventually make it a bit more explicit (by variable naming or comment adding) anyhow.

Reply:
Sorry it wasn't very clear, because the suite setup is happening at the beginning of the file. Here and in the other test case `runTx` and `runCall` are failing because I've replaced them with a mock that simply calls the callback with an error.

I added comments to explain this in the source file as well.